domain/learningandworkprogress/src/main/kotlin/uk/gov/justice/digital/hmpps/domain/learningandworkprogress/induction/PreviousWorkExperiences.kt

```diff
@@ -49,3 +49,9 @@ enum class WorkExperienceType {
   CLEANING_AND_MAINTENANCE,
   OTHER,
 }
+
+enum class HasWorkedBeforeDomain {
```

Comment:
Minor comment on this (and the enum at the entity level) .... is there a reason why you've suffixed the classname with `Domain` and `Entity`?
I don't particularly mind, but if it was to differentiate the classes in some way, their package hierarchy should do that sufficiently for us? 🤔 

If however it was to do with import clashes in classes like the mappers, where you are mapping to/from `HasWorkedBefore` as an API type to a domain type for example .... kotlin has you covered here 😁 
With kotlin you can do import aliasing, so you'd be able to do this:
```
import uk.gov.justice.digital.hmpps.domain.learningandworkprogress.induction.HasWorkedBefore as HasWorkedBeforeDomain
...
import uk.gov.justice.digital.hmpps.educationandworkplanapi.resource.model.HasWorkedBefore
```

(I don't mind though if you want to name them this way for explicitness 👍 )

Reply:
Yeah it was to distinguish them for imports really, there is a clash in a mapper but also when importing into tests, etc. you have to pick 1 of 3. TBH if there wasn't precedent for duplicating them I'd just use the auto-generated one in all 3 cases given they have the same values on only specialise if there is some sort of drift.

Reply:
Hopefully you wouldnt be able to do that; not in the case of the domain anyway.
The way we've organised the modules and the dependencies etc, then domain module can't see the main application so therefore wouldn't be able to see & import the classes from the main module (either the generated or the entity classes)

This was very deliberate. The reason the domain is separate is that some of this stuff doesn't actually belong in our repo at all! Timeline for example ... it would be much better if there were a common/central HMPPS timeline service; and maybe all we'd need to do was import a jar as a dependency if we wanted to do "timeline" stuff
Likewise goals and steps .... When they move to OnePlan we wont own that domain, but we will (almost certainly) still need some REST API endpoints relating to them in some way.

Anyway, 3 layers (albeit with duplication) & mappers FTW 😁 👍